game/src/ecs/components/ai/AIComponent.java

```diff
@@ -30,17 +37,15 @@ public AIComponent(Entity entity) {
         idleAI = new RadiusWalk(5);
         transition = new RangeTransition(1.5f);
         fightAI =
-                new IFightAI() {
-                    @Override
-                    public void fight(Entity entity) {
-                        System.out.println("TIME TO FIGHT!");
-                        // todo replace with melee skill
-                    }
+                entity1 -> {
```

Comment:
@cagix kannst du mir da erklären? War ein Vorschlag von IntelliJ und der hat mich irgendwie sehr verwirrt und ich wollte ihn teilen.

mich irritiert dieses `entity1`

Reply:
hmmm. auf der rechten seite steht ein lambda-ausdruck: `entity1` ist der parameter, und hinter dem `->` kommt der "funktionskörper". geschweifte klammern, weil du keine einzelne expression hast, sondern ein statement (`println()`) und ij vermutlich wg. des kommentars noch weitere statements erwartet.

die frage ist, warum die zuweisung und was bedeutet die linke seite? `fightAI` ist vom typ `IFightAI`, und das ding ist ein "funktionales interface", hat also genau eine abstrakte methode `void fight(Entity entity);`. 

d.h. die ganze rechte seite ist eine anonyme "on-the-fly"-implementierung für `void fight(Entity entity);`. oder andersherum, du kannst anschließend `fightAI(entity)` aufrufen :)

Reply:
> entity1 ist der parameter,

Danke. Das ging mir irgendwie nicht in die Birne....

Reply:
danke für das beispiel :) ... das baue ich in die vorlesung als brainboggle ein :)

Reply:
ich lach mich schlapp :) ... grad wollte ich fragen, was da vorher stand, also was ij da ersetzt hat. und dann sehe ich den roten bereich ;) ... 

aber letztlich ist es genau das. vorher hattest du dort eine anonyme klasse, die direkt instanziiert wurde - also ein objekt, welches vom typ her das interface erfüllt und diese eine methode anbietet. jetzt (nachher) hast du das auch, aber hier als lambda und es macht den fokus auf diese eine methode im interface deutlich.

(diese ersetzung klappt aber tatsächlich nur bei funktionalen interfaces, also wenn es nur genau eine abstrakte methode im interface gibt.)